contracts/extension/NFTMetadata.sol

```diff
@@ -15,25 +17,25 @@ abstract contract NFTMetadata is INFTMetadata {
     /// @notice Sets the metadata URI for a given NFT.
     function _setTokenURI(uint256 _tokenId, string memory _uri) internal virtual {
         require(bytes(_uri).length > 0, "NFTMetadata: empty metadata.");
-        require(!_URIFrozen[_tokenId], "NFTMetadata: metadata is frozen.");
         _tokenURI[_tokenId] = _uri;
 
         emit MetadataUpdate(_tokenId);
     }
 
     /// @notice Sets the metadata URI for a given NFT.
     function setTokenURI(uint256 _tokenId, string memory _uri) public virtual {
-        require(_canSetMetadata(_tokenId), "NFTMetadata: not authorized to set metadata.");
+        require(_canSetMetadata(), "NFTMetadata: not authorized to set metadata.");
+        require(!_URIFrozen, "NFTMetadata: metadata is frozen.");
         _setTokenURI(_tokenId, _uri);
     }
 
-    function freezeTokenURI(uint256 _tokenId) public virtual {
-        require(_canFreezeMetadata(_tokenId), "NFTMetadata: not authorized to freeze metdata");
-        _URIFrozen[_tokenId] = true;
+    function freezeMetadata() public virtual {
+        require(_canFreezeMetadata(), "NFTMetadata: not authorized to freeze metdata");
+        _URIFrozen = true;
```

Comment:
Let's emit an event here to signify that metadata is now frozen. I saw you mention `PermanentURI` recognized by OpenSea. We should emit this event, or follow some EIP for this event if it exists.

Reply:
Two FYIs:

1. I see one relevant EIP - ERC3569. It is in draft and has had no comment or updates since it was submitted in 2021. I don't think it is worth using as it quite different than what we currently have.
2. The OS supported event is one event per token, and won't work for all of the contracts metadata being frozen at once. 

That said, I don't mind adding in an event that indicates metadata is frozen - and possibly at some point in the future there may be one that supported by marketplaces.